src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -144,9 +148,12 @@
             SslEngineSource sslEngineSource,
             boolean authenticateClients,
             ChannelPipeline pipeline,
-            GlobalTrafficShapingHandler globalTrafficShapingHandler) {
+            GlobalTrafficShapingHandler globalTrafficShapingHandler,
+            Channel channel) {
         super(AWAITING_INITIAL, proxyServer, false);
 
+        this.channel = channel;
```

Comment:
initial the channel is assigned on `channelRegistered` which happens a bit later but we need the channel right now to create global state wrapper event loop